### PR TITLE
fix: added checks to prevent errors when parameter gradients were disabled 

### PR DESCRIPTION
**Pull request by PaulCCCCCCH**
After producing visualizations, these hooks will remain registered with the model, and when a user tries to backprop any parameter with gradient disabled, it will cause problems. Specifically, `grad_in[0]` in `_record_gradients` will be None, and `_record_output` may not be triggered. We can remind users to unregister those hooks after using flashtorch, but a simple thing that we could do is to skip those hooks by based on some checks.

### CODE REVIEW

**Comment by PaulCCCCCCH** [comment]
Basically I was using `flashtorch` and [`python-influence-function`](https://github.com/nimarb/pytorch_influence_functions) at the same time to provide model insights for my project, but I found that every time I ran influence function after flashtorch visualization, it influence function would fail. I spent a lot of time debugging influence function instead, not realizing the hooks here were the cause.
